src/Thentos/Frontend/Handlers.hs

```diff
@@ -110,20 +111,90 @@ userCreateConfirm = do
             logger DEBUG "no token"
             blaze (errorPage "finializing registration failed: token is missing.")
 
-runWithUserClearance :: (ThentosClearance -> FH ()) -> FH ()
+userUpdate :: FH ()
+userUpdate = runWithUserClearance $ \ clearance uid -> do
+    (userView, result) <- runForm "update" userUpdateForm
+    (emailView, _) <- runForm "update_email" emailUpdateForm
+    (pwView, _) <- runForm "update_password" passwordUpdateForm
+    case result of
+        Nothing -> blaze $ userUpdatePage userView emailView pwView
+        Just fieldUpdates -> do
+            result' <- update $ UpdateUserFields uid fieldUpdates clearance
+            case result' of
+                Right () -> blaze "User data updated!"
+                Left e -> logger INFO (show e) >> blaze (errorPage "user update failed.")
+
+passwordUpdate :: FH ()
+passwordUpdate = runWithUserClearance $ \ clearance uid -> do
+    (passwordView, result) <- runForm "update_password" passwordUpdateForm
+    (userView, _) <- runForm "update" userUpdateForm
+    (emailView, _) <- runForm "update_email" emailUpdateForm
+    case result of
+        Nothing -> blaze $ userUpdatePage userView emailView passwordView
+        Just (oldPw, newPw) -> do
+            result' <- snapRunAction' clearance $ changePassword uid oldPw newPw
+            case result' of
+                Right () -> blaze "Password Changed!"
+                Left e -> logger INFO (show e) >> blaze (errorPage "user update failed.")
+
+emailUpdate :: FH ()
+emailUpdate = runWithUserClearance $ \ clearance uid -> do
+    (passwordView, _) <- runForm "update_password" passwordUpdateForm
+    (userView, _) <- runForm "update" userUpdateForm
+    (emailView, result) <- runForm "update_email" emailUpdateForm
+    case result of
+        Nothing -> blaze $ userUpdatePage userView emailView passwordView
+        Just newEmail -> do
+            result' <- snapRunAction' clearance $ requestUserEmailChange uid newEmail
+            case result' of
+                Right token -> do
+                    config :: ThentosConfig <- gets (^. cfg)
+                    feConfig <- gets (^. frontendCfg)
+                    liftIO $ sendEmailChangeConfirmMail (Tagged $ config >>. (Proxy :: Proxy '["smtp"]))
```

Comment:
send out emails in `Thentos.Api`?  the benefit would be that the backend does not have to duplicate that code if email update is every triggered via rest.  (the idea is that thentos can be accessed both via the snap api and -- alternatively -- entirely via rest.)